### PR TITLE
Switch create validation handled

### PR DESCRIPTION
**Pull request by selvakumarnms**
1. Switch create validation for discovery protocol name and id
2. Switch delete support based on name
3. Switch delete validation handled for multiple same names
4. Implemented DB helper to support above fixes.

### CODE REVIEW

File: baremetal_network_provisioning/ml2/extensions/bnp_switch.py [hunk 1]
```diff
@@ -160,18 +161,37 @@ def show(self, request, id, **kwargs):
...
+            raise webob.exc.HTTPConflict(
+                _("Multiple switches matches found "
+                  "for name %s, use an ID to be more specific.") % id)
+        if isinstance(switch, list) and len(switch) == 1:
+            portmap = db.get_bnp_switch_port_map_by_switchid(context,
+                                                             switch[0].id)
+            port_prov = switch[0].port_prov
+        else:
+            portmap = db.get_bnp_switch_port_map_by_switchid(context, id)
+            port_prov = switch['port_prov']
```
Comment (Mouli-HPE) [inline baremetal_network_provisioning/ml2/extensions/bnp_switch.py]: This else option will never be hit.
as >1 and ==0 case already handled above.

**Comment by Mouli-HPE** [comment]
Line 182 in bnp_switch.py
else option will never be hit.
as >1 and ==0 case already handled.

File: baremetal_network_provisioning/ml2/extensions/bnp_switch.py (hunk 1 above)
Comment (selvakumarnms) [inline baremetal_network_provisioning/ml2/extensions/bnp_switch.py]: It will hit in this case of id.

**Comment by hp-networking** [comment]
Looks good.

**Pull request merged**